include/boost/geometry/algorithms/simplify.hpp

```diff
@@ -472,14 +474,29 @@ public :
                 continue;
             }
 
-            std::rotate_copy(boost::begin(ring), range::pos(ring, index),
-                             boost::end(ring), rotated.begin());
+            // Do not duplicate the closing point
+            auto rot_end = boost::end(ring);
+            std::size_t rot_index = index;
+            if (is_closed && size > 1)
+            {
+                --rot_end;
+                if (rot_index == size - 1) { rot_index = 0; }
+            }
+
+            std::rotate_copy(boost::begin(ring), range::pos(ring, rot_index),
+                             rot_end, std::back_inserter(rotated));
 
             // Close the rotated copy
-            rotated.push_back(range::at(ring, index));
+            rotated.push_back(range::at(ring, rot_index));
 
             simplify_range<0>::apply(rotated, out, max_distance, impl, strategies);
 
+            // Open output if needed
+            if (! is_closed && boost::size(out) > 1)
+            {
+                range::pop_back(out);
+            }
```

Comment:
Maybe I get it wrongly, but above you rotate with one point less (fine) - and now you pop another one. Is that not one too much?

Reply:
These are two different issues.

The point above is removed only for closed rings. Otherwise when they are rotated then two equal points (beginning and closing) are moved somewhere into the middle of the ring.

After the rotation an additional closing point is pushed back, always, even for open rings. This is why here, for open rings this additional point is removed.

Reply:
An alternative to `pop_back()` would be not copying the last point for open output ring inside the `simplify_range<0>::apply()`.